### PR TITLE
feat: add category legend

### PR DESCRIPTION
**Pull request by ondrej-hosak**

### CODE REVIEW

**Comment by lksv** [comment]
LGTM, zitra vyzkousim a mergnu.

File: app/controllers/categories_controller.rb [hunk 1]
```diff
@@ -11,4 +11,8 @@ def options
     res = res.find_all { |c| c[:text] =~ /\b#{Regexp.escape(params[:query].to_s)}/ }
     render json: res
   end
+
+  def index
+    render json: Category::CATEGORY_MAPPING.values
```
Comment (lksv) [inline app/controllers/categories_controller.rb]: prefer to put also keys. 
E.g.

``` ruby
Category::CATEGORY_MAPPING.map { |k,v| v[:key] = k }
```

I know where to could I use it just now:)
uz ted vim, kde bych to pouzil:)

File: app/views/map/index.html.haml [hunk 2]
```diff
@@ -114,6 +114,7 @@
   .center
     Pro zobrazení událostí musíte přiblížit mapu.
 
+= javascript_include_tag 'categories_legend.js'
```
Comment (lksv) [inline app/views/map/index.html.haml]: I would prefer to put it to assets (application.js) event it is used only in map.

File: config/routes.rb [hunk 3]
```diff
@@ -34,6 +34,7 @@
   get 'tiles/:z/:x/:y'  =>  'events#tiles', as: :tiles
   get 'public/tiles/:z/:x/:y'  =>  'events#public_tiles', as: :public_tiles
   get 'categories/options' => 'categories#options', as: :categories_options
+  get 'categories' => 'categories#index', as: :categories_index
```
Comment (lksv) [inline config/routes.rb]: there are two categories routes, so prefer

```
resources :categories, only: [:index] do
   get :options, on: :collection
end
```

**Comment by ondrej-hosak** [comment]
@lksv please review updated version

**Comment by lksv** [comment]
perfekt!

**Pull request merged**